src/lib/y2storage/disk_analyzer.rb

```diff
@@ -254,105 +255,131 @@ def release_name(filesystem)
-    # our installation medium - the medium we just booted and started the
-    # installation from.
+    # Checks whether a device contains an installation repository
     #
-    # The method to achieve this is to try to mount the filesystem and check
-    # if there is a file /control.xml and compare its content to the
-    # /control.xml in the inst-sys. We cannot simply compare device
-    # minor/major IDs since the inst-sys is in a RAM disk (copied from the
-    # installation medium).
+    # For all possible names of the given device, it is checked if any of that
+    # names is included in the URI of an installation repository (see
+    # {#repositories_devices}). Note that the names of all devices inside the
+    # given device are considered as names of the given device (see #{device_names}),
+    # (e.g., when a disk contains a partition being used as LVM PV, the names of the
+    # LVM LVs are considered as names of the disk).
     #
-    # @param device [#name] device to check
-    # @return [Boolean] 'true' if the volume is an installation volume
-    def installation_volume?(device)
-      log.info("Checking if #{device.name} is an installation medium")
-      return false unless device.filesystem
-      fs = ExistingFilesystem.new(device.filesystem)
-      log.info("#{device.name} is an installation medium") if fs.installation_medium?
-      fs.installation_medium?
+    # @param device [BlkDevice]
+    # @return [Boolean]
+    def contain_installation_repository?(device)
+      device_names(device).any? { |n| repositories_devices.include?(n) }
+    end
+
+    # All possible device names of a device
+    #
+    # Device names includes the kernel name and all udev names given by libstorage-ng.
+    # Moreover, it includes the names of all devices inside the given device
+    # (e.g., names of partitions inside a disk). Note that when a device contains a
+    # partition being used as LVM PV, the names of the LVM LVs are considered as names
+    # of the device.
+    #
+    # @param device [BlkDevice]
+    # @return [Array<String>]
+    def device_names(device)
+      devices = all_devices_from_device(device)
+
+      names = devices.map { |d| d.udev_full_all.prepend(d.name) }
+      names.flatten.compact.uniq
+    end
+
+    # All blk devices defined from a device, including the given device
+    # (e.g., a disk and all its partitions)
+    #
+    # Note that when a device contains a partition being used as LVM PV, all LVM LVs are included.
+    #
+    # @param device [BlkDevice]
+    # @return [Array<BlkDevice>]
+    def all_devices_from_device(device)
+      devices = device.descendants.select { |d| d.is?(:blk_device) }
+      devices.prepend(device)
+    end
+
+    # Device names indicated in the URI of the installation repositories
+    #
+    # @see #local_repositories
+    #
+    # @return [Array<String>]
+    def repositories_devices
+      return @repositories_devices if @repositories_devices
```

Comment:
Is this memoization worth it? Does it really save time or resources?

Just asking because I'm concerned about situations in which the user adds a new local repository (going back and forward in the installation process). Quite a corner case, I know.

Reply:
As commented in IRC, this values are recalculated when going back and forward, due to the system is repobed and DiskAnalyzer is created again.

Reply:
Actually there would be a corner case if the repositories can be added in a step that comes after the proposal (let's say the software selection that can be triggered in the summary screen) and then the user goes back to the guided setup (without reprobing).

But I admit that's really a wild corner case and the user deserves all the harm. :wink: